src/main/java/net/mcreator/element/converter/fv22/BlockLightOpacityFixer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.converter.fv22;
+
+import com.google.gson.JsonElement;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.converter.IConverter;
+import net.mcreator.element.types.Block;
+import net.mcreator.workspace.Workspace;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class BlockLightOpacityFixer implements IConverter {
+	private static final Logger LOG = LogManager.getLogger(BlockLightOpacityFixer.class);
+
+	@Override
+	public GeneratableElement convert(Workspace workspace, GeneratableElement input, JsonElement jsonElementInput) {
+		Block block = (Block) input;
+		try {
+			if (block.blockBase == null || block.blockBase.equals("")) {
+				block.lightOpacity /= 16;
+			}
+			else if (block.blockBase.equals("Leaves")) {
+				block.lightOpacity = 1;

```

Comment:
I think there is no need for this one as you already take care of this in templates

Reply:
The method is always overriden with the leaves base, so this is required (the opacity in game was always 1, so it shouldn't affect existing leaves).

Reply:
But in case leaves are used this parameter is not checked so I don't think you need to set it here. New blocks with leaves base won't have 1 either I think.

Reply:
This parameter is checked when the leaves base is selected, and when creating a new block it's set to 1

Reply:
I think you should then disable opacity field when leaves block base is used.

Reply:
I don't see why. The option works as intended with the leaves base, and it's probably the best block base to use this property with

Reply:
But leaves should always have this value 1 if I understood correctly or it looks stange

Reply:
Not really, the vanilla leaves set it to 1, but it's fine to change it with custom leaves

Reply:
I see, thank you for the clarification :)